src/core/Fingerprint.cpp

```diff
@@ -0,0 +1,228 @@
+#include "Fingerprint.hpp"
+#include "../helpers/Log.hpp"
+#include "../config/ConfigManager.hpp"
+
+#include <filesystem>
+#include <unistd.h>
+#include <pwd.h>
+
+#include <cstring>
+#include <thread>
+
+static const auto FPRINT        = "net.reactivated.Fprint";
+static const auto DEVICE        = "net.reactivated.Fprint.Device";
+static const auto MANAGER       = "net.reactivated.Fprint.Manager";
+static const auto LOGIN_MANAGER = "org.freedesktop.login1.Manager";
+static const auto RETRY_MESSAGE = "Could not match fingerprint. Try again.";
+
+CFingerprint::CFingerprint() {
+    static auto* const PFINGERPRINTREADY   = (Hyprlang::STRING*)(g_pConfigManager->getValuePtr("general:fingerprint_ready_message"));
+    m_sFingerprintReady                    = *PFINGERPRINTREADY;
+    static auto* const PFINGERPRINTPRESENT = (Hyprlang::STRING*)(g_pConfigManager->getValuePtr("general:fingerprint_present_message"));
+    m_sFingerprintPresent                  = *PFINGERPRINTPRESENT;
+    static auto* const PENABLEFINGERPRINT  = (Hyprlang::INT* const*)g_pConfigManager->getValuePtr("general:enable_fingerprint");
+    m_bEnabled                             = **PENABLEFINGERPRINT;
+}
+
+static void fingerprintCheckCallback(std::shared_ptr<CTimer> self, void* data) {
+    if (g_pFingerprint->isAuthenticated())
+        g_pHyprlock->unlock();
+}
+
+std::optional<std::thread> CFingerprint::start() {
+    if (!m_bEnabled)
+        return {};
+    return std::thread([this]() {
```

Comment:
absolutely do _not_ multithread here. Get the pollfd from sdbus-cpp and integrate it within the main event loop (see xdph: https://github.com/hyprwm/xdg-desktop-portal-hyprland/blob/fb9c8d665af0588bb087f97d0f673ddf0d501787/src/core/PortalManager.cpp#L270)

Reply:
Done. I was originally trying to maintain more consistency with CAuth.

It makes things a bit more complicated, but I kept it so that the dbus connection is only initiated/listened to if fingerprint auth is enabled. I'd be happy to change it to always create the connection if you'd rather that.